apps/openmw/mwmechanics/spellcasting.cpp

```diff
@@ -1199,10 +1199,10 @@ namespace MWMechanics
         return false;
     }
 
-    void adjustDynamicStat(CreatureStats& creatureStats, int index, float magnitude)
+    void adjustDynamicStat(CreatureStats& creatureStats, int index, float magnitude, bool capAt0 = true)
     {
         DynamicStat<float> stat = creatureStats.getDynamic(index);
-        stat.setCurrent(stat.getCurrent() + magnitude, index == 2);
+        stat.setCurrent(stat.getCurrent() + magnitude, !capAt0);
```

Comment:
Use parameter name from [this](https://github.com/OpenMW/openmw/blob/d7989ac7782272628745eb7d920dc477d57c6809/apps/openmw/mwmechanics/stat.hpp#L101) method (`allowDecreaseBelowZero`) to remove boolean conversions.